### PR TITLE
Fix spelling mistakes in the docs

### PR DESCRIPTION
**Pull request by Tempate**
Fix a few misspells in the Discovery Settings, Shared Memory Transport, and Typical Use Cases sections.

### CODE REVIEW

File: docs/fastdds/discovery/simple.rst [hunk 1]
```diff
@@ -119,7 +119,7 @@ Initial peers
 According to the `RTPS standard <https://www.omg.org/spec/DDSI-RTPS/2.2/PDF>`_ (Section 9.6.1.1), each
 |RTPSParticipant-api|
 must listen for incoming Participant Discovery Protocol (PDP) discovery metatraffic in two different ports, one linked
-with a multicast address, and another one linked to a unicast address.
+to a multicast address and another one linked to a unicast address.
```
Comment (JesusPoderoso) [inline docs/fastdds/discovery/simple.rst:122]:
```suggestion
to a multicast address, and another one linked to a unicast address.
```
We normally use a comma before an "and" to splits two independent clauses.

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]

**Comment by JesusPoderoso** [comment]
Please, rebase this branch against master too

File: docs/fastdds/discovery/simple.rst (hunk 1 above)
Comment (Tempate) [reply docs/fastdds/discovery/simple.rst:122]: I believe the clauses aren't independent. The sentence is clarifying the two different types of ports, so both need to stay together. The sentence "another one linked to a unicast address" could not stand alone, and therefore isn't independent.

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**